website/announcements/middleware.py

```diff
@@ -22,3 +23,21 @@ def __call__(self, request):
         response.set_cookie("closed-announcements", encode_closed_announcements(closed_announcements))
 
         return response
+
+
+class AnnouncementMiddleware:
+    """Announcement Middleware."""
+
+    def __init__(self, get_response):
+        """Initialize Announcement Middleware."""
+        self.get_response = get_response
+
+    def __call__(self, request):
+        """Call app announcement function if they exist for gathering all announcements."""
+        announcements = []
+        for app in apps.get_app_configs():
+            if hasattr(app, "announcements"):
```

Comment:
Maybe check if `announcements` is not only an attribute but also a function that can be called